tests/lib.rs

```diff
@@ -876,3 +876,304 @@ Error codes:
+        /// speed of cmd
+        speed: u8,
+
+        #[argh(option)]
+        /// direction
+        direction: String,
+    }
+
+    let actual = Cmd::redact(&["<<<arg0>>>"], &["5", "--direction", "north"]).unwrap();
+    assert_eq!(actual, &["<<<arg0>>>", "speed", "--direction"]);
+}
+
+#[test]
+fn redact_positional_optional_option() {
+    #[derive(FromArgs, Debug)]
+    /// Short description
+    struct Cmd {
+        #[argh(positional)]
+        /// speed of cmd
+        speed: u8,
+
+        #[argh(option)]
+        /// direction
+        direction: Option<String>,
+    }
+
+    let actual = Cmd::redact(&["<<<arg0>>>"], &["5"]).unwrap();
+    assert_eq!(actual, &["<<<arg0>>>", "speed"]);
+}
+
+#[test]
+fn redact_subcommand() {
+    #[derive(FromArgs, Debug)]
+    /// Short description
+    struct Cmd {
+        #[argh(positional)]
+        /// speed of cmd
+        speed: u8,
+
+        #[argh(subcommand)]
+        /// means of transportation
+        means: MeansSubcommand,
+    }
+
+    #[derive(FromArgs, Debug)]
+    /// Short description
+    #[argh(subcommand)]
+    enum MeansSubcommand {
+        Walking(WalkingSubcommand),
+        Biking(BikingSubcommand),
+        Driving(DrivingSubcommand),
+    }
+
+    #[derive(FromArgs, Debug)]
+    #[argh(subcommand, name = "walking")]
+    /// Short description
+    struct WalkingSubcommand {
+        #[argh(option)]
+        /// a song to listen to
+        music: String,
+    }
+
+    #[derive(FromArgs, Debug)]
+    #[argh(subcommand, name = "biking")]
+    /// Short description
+    struct BikingSubcommand {}
+    #[derive(FromArgs, Debug)]
+    #[argh(subcommand, name = "driving")]
+    /// short description
+    struct DrivingSubcommand {}
+
+    let actual = Cmd::redact(&["<<<arg0>>>"], &["5", "walking", "--music", "Bach"]).unwrap();
+    assert_eq!(actual, &["<<<arg0>>>", "speed", "walking", "--music"]);
+}
+
+#[test]
+fn redact_subcommand_with_space_in_name() {
+    #[derive(FromArgs, Debug)]
+    /// Short description
+    struct Cmd {
+        #[argh(positional)]
+        /// speed of cmd
+        speed: u8,
+
+        #[argh(subcommand)]
+        /// means of transportation
+        means: MeansSubcommand,
+    }
+
+    #[derive(FromArgs, Debug)]
+    /// Short description
+    #[argh(subcommand)]
+    enum MeansSubcommand {
+        Walking(WalkingSubcommand),
+        Biking(BikingSubcommand),
+    }
+
+    #[derive(FromArgs, Debug)]
+    #[argh(subcommand, name = "has space")]
+    /// Short description
+    struct WalkingSubcommand {
+        #[argh(option)]
+        /// a song to listen to
+        music: String,
+    }
+
+    #[derive(FromArgs, Debug)]
+    #[argh(subcommand, name = "biking")]
+    /// Short description
+    struct BikingSubcommand {}
+
+    let actual = Cmd::redact(&["<<<arg0>>>"], &["5", "has space", "--music", "Bach"]).unwrap();
+    assert_eq!(actual, &["<<<arg0>>>", "speed", "has space", "--music"]);
+}
+
+#[test]
+fn redact_produces_help() {
```

Comment:
I don't see any test cases where `redact()` returns an `EarlyExit` with an `Err` (did I miss it?)

Reply:
I added some, thanks for catching this.